py_wrap_generator.py

```diff
@@ -0,0 +1,2101 @@
+				if candidate != None:
+					enums.append(candidate)
+					debug("\tFound enum \"" + candidate.name + "\" in namespace " + concat_namespace(namespaces),2)
+				elif class_ != None and class_[1] == 1:
+					candidate = WConstructor.from_string(ugly_line, source.name, class_[0], i)
+					if candidate != None:
+						debug("\t\tFound constructor of class \"" + class_[0].name + "\" in namespace " + concat_namespace(namespaces),2)
+						class_[0].found_constrs.append(candidate)
+					else:
+						candidate = WMember.from_string(ugly_line, source.name, class_[0], i, concat_namespace(namespaces))
+						if candidate != None:
+							if type(candidate) == list:
+								for c in candidate:
+									debug("\t\tFound member \"" + c.name + "\" of class \"" + class_[0].name + "\" of type \"" + c.wtype.name + "\"", 2)
+								class_[0].found_vars.extend(candidate)
+							else:
+								debug("\t\tFound member \"" + candidate.name + "\" of class \"" + class_[0].name + "\" of type \"" + candidate.wtype.name + "\"", 2)
+								class_[0].found_vars.append(candidate)
+
+			j = i
+			line = unpretty_string(line)
+			while candidate == None and j+1 < len(source_text) and  line.count(';') <= 1 and line.count("(") >= line.count(")"):
+				j = j + 1
+				line = line + "\n" + unpretty_string(source_text[j])
+				if class_ != None:
+					candidate = WFunction.from_string(ugly_line, source.name, class_[0], i, concat_namespace(namespaces))
+				else:
+					candidate = WFunction.from_string(ugly_line, source.name, None, i, concat_namespace(namespaces))
+				if candidate != None and candidate.name.find("::") == -1:
+					if class_ == None:
+						debug("\tFound unowned function \"" + candidate.name + "\" in namespace " + concat_namespace(namespaces),2)
+						unowned_functions.append(candidate)
+					else:
+						debug("\t\tFound function \"" + candidate.name + "\" of class \"" + class_[0].name + "\" in namespace " + concat_namespace(namespaces),2)
+						class_[0].found_funs.append(candidate)
+					continue
+				candidate = WEnum.from_string(line, concat_namespace(namespaces), i)
+				if candidate != None:
+					enums.append(candidate)
+					debug("\tFound enum \"" + candidate.name + "\" in namespace " + concat_namespace(namespaces),2)
+					continue
+				if class_ != None:
+					candidate = WConstructor.from_string(line, source.name, class_[0], i)
+					if candidate != None:
+						debug("\t\tFound constructor of class \"" + class_[0].name + "\" in namespace " + concat_namespace(namespaces),2)
+						class_[0].found_constrs.append(candidate)
+						continue
+		if candidate != None:
+			while i < j:
+				i = i + 1
+				line = source_text[i].replace("YOSYS_NAMESPACE_BEGIN", "                    namespace Yosys{").replace("YOSYS_NAMESPACE_END","                    }")
+				ugly_line = unpretty_string(line)
+				if len(namespaces) != 0:
+					namespaces[-1] = (namespaces[-1][0], namespaces[-1][1] + ugly_line.count("{
```

Comment:
I think this should be USING_YOSYS_NAMESPACE in case the Yosys namespace is changed.

Reply:
USING_YOSYS_NAMESPACE will not do, since I only need the name of the namespace. I introduced a new define in kernel/yosys.h that only contains the namespace name and used that one. It is near the other defines that need to be changed in case the namespace changes.
0774a500d4a03104787e6514c69ff4995633a65f